### PR TITLE
Fix event spread in PriceField

### PR DESCRIPTION
**Pull request by taniotanio7**
I want to merge this change because it fixes an issue introduced in #1954 that caused to create incorrect `e.target` due to object spread instead of modyfing the `e.target.value`

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-6312 All price fields should be 2 decimals](https://app.clickup.com/t/2549495/SALEOR-6312)

**Review by dominik-zeglen — APPROVED** [review]

File: src/components/PriceField/PriceField.tsx [hunk 1]
```diff
@@ -71,21 +71,20 @@ export const PriceField: React.FC<PriceFieldProps> = props => {
...
   const handleChange: React.ChangeEventHandler<HTMLInputElement> = e => {
-    let newValue = e.target.value;
-    const splitCharacter = findPriceSeparator(newValue);
+    const splitCharacter = findPriceSeparator(e.target.value);
     const [integerPart, decimalPart] = e.target.value.split(splitCharacter);
 
     if (maxDecimalLength === 0 && decimalPart) {
       // this shouldn't happen - decimal character should be ignored
-      newValue = integerPart;
+      e.target.value = integerPart;
```
Comment (Droniu) [inline src/components/PriceField/PriceField.tsx:79]: This value should not be mutated. If we can't deep clone `e` with vanilla JS I suggest using lodash

File: src/components/PriceField/PriceField.tsx [hunk 2]
```diff
@@ -71,21 +71,20 @@ export const PriceField: React.FC<PriceFieldProps> = props => {
...
     }
 
     if (decimalPart?.length > maxDecimalLength) {
       const shortenedDecimalPart = decimalPart.slice(0, maxDecimalLength);
-      newValue = `${integerPart}${splitCharacter}${shortenedDecimalPart}`;
+      e.target.value = `${integerPart}${splitCharacter}${shortenedDecimalPart}`;
```
Comment (Droniu) [inline src/components/PriceField/PriceField.tsx:84]: Same here.

**Review by Droniu — CHANGES_REQUESTED** [review]

File: src/components/PriceField/PriceField.tsx (hunk 1 above)
Comment (dominik-zeglen) [reply src/components/PriceField/PriceField.tsx:79]: Well it works, so i guess it's okay to mutate it

From the lodash docs:

Note: This method is loosely based on the [structured clone algorithm](https://mdn.io/Structured_clone_algorithm) and supports cloning arrays, array buffers, booleans, date objects, maps, numbers, Object objects, regexes, sets, strings, symbols, and typed arrays. The own enumerable properties of arguments objects are cloned as plain objects. An empty object is returned for uncloneable values such as error objects, functions, DOM nodes, and WeakMaps.

We're talking about DOM node.

**Review by Droniu — APPROVED** [review]

**Pull request merged**